ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -74,48 +74,70 @@ ParallelConfig determine_parallel_config(
     uint32_t max_num_cores = device_grid_size[0] * device_grid_size[1];
 
     auto calculate_num_cores_nhw = [&]() {
-        uint32_t num_cores_nhw =
-            height_sharding
-                ? find_closest_largest_divisor(conv_out_2d_matrix_height_ntiles, max_num_cores)
-                : find_closest_largest_divisor_with_num_padding(conv_out_2d_matrix_height_ntiles, device_grid_size[0]);
-        return num_cores_nhw;
+
+        if(shard_layout == TensorMemoryLayout::HEIGHT_SHARDED) {
+            return find_closest_largest_divisor(conv_out_2d_matrix_height_ntiles, max_num_cores);
+        } else if(shard_layout == TensorMemoryLayout::BLOCK_SHARDED) {
+            return find_closest_largest_divisor_with_num_padding(conv_out_2d_matrix_height_ntiles, device_grid_size[0]);
+        } else if(shard_layout == TensorMemoryLayout::WIDTH_SHARDED) {
+            return 1u;
+        }
+
+        TT_FATAL("Invalid Shard Layout",shard_layout);
```

Comment:
Imagine that you are just using TT-NN, running conv and see an error. It would tell me absolutely nothing. 
Give me more context. Something like "Conv operation only supports ....... but <x> is passed."

Reply:
Fixed.